### PR TITLE
MULE-11133 Create initial flow performance test case

### PR DESCRIPTION
**Pull request by dfeist**

### CODE REVIEW

File: core/src/main/java/org/mule/runtime/core/construct/AbstractPipeline.java [hunk 1]
```diff
@@ -236,7 +237,7 @@ protected void doInitialise() throws MuleException {
 
         @Override
         public Event process(final Event event) throws MuleException {
-          if (isTransactionActive()) {
+          if (isTransactionActive() || processingStrategy == SYNCHRONOUS_PROCESSING_STRATEGY_INSTANCE) {
```
Comment (marianogonzalez) [inline core/src/main/java/org/mule/runtime/core/construct/AbstractPipeline.java]: I think the processingStrategy itself should have a isSynchronous() method. These kind of comparisons are pretty fragile and hard to test.

File: tests/performance/src/test/java/org/mule/test/core/flow/FlowPerformanceTestCase.java [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+
+  @Test
+  @PerfTest(duration = 15000, threads = 1, warmUp = 5000)
+  public void fluxPush() throws Exception {
+    CountDownLatch latch = new CountDownLatch(ITERATIONS);
+    for (int i = 0; i < ITERATIONS; i++) {
+      Event event = Event.builder(DefaultEventContext.create(flow, TEST_CONNECTOR))
+          .message(InternalMessage.of(TEST_PAYLOAD)).build();
+      fluxSink.accept(event);
+      Mono.from(event.getContext()).doOnNext(e -> latch.countDown()).subscribe();
```
Comment (marianogonzalez) [inline tests/performance/src/test/java/org/mule/test/core/flow/FlowPerformanceTestCase.java]: `e -> latch.countDown()` -> `latch::countDown`

File: tests/performance/src/test/java/org/mule/test/core/flow/FlowPerformanceTestCase.java [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+  @PerfTest(duration = 15000, threads = 1, warmUp = 5000)
+  public void fluxPull() throws Exception {
+    AtomicInteger requests = new AtomicInteger();
+    CountDownLatch latch = new CountDownLatch(ITERATIONS);
+
+    Flux.<Event>generate(eventSynchronousSink -> {
+      if (requests.getAndIncrement() < ITERATIONS) {
+        Event event = Event.builder(DefaultEventContext.create(flow, TEST_CONNECTOR))
+            .message(InternalMessage.of(TEST_PAYLOAD)).build();
+        Mono.from(event.getContext()).doOnNext(e -> latch.countDown())
```
Comment (marianogonzalez) [inline tests/performance/src/test/java/org/mule/test/core/flow/FlowPerformanceTestCase.java]: idem

File: tests/performance/src/test/java/org/mule/test/core/flow/FlowPerformanceTestCase.java [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+          .message(InternalMessage.of(TEST_PAYLOAD)).build();
+      fluxSink.accept(event);
+      Mono.from(event.getContext()).doOnNext(e -> latch.countDown()).subscribe();
+    }
+    latch.await(LOCK_TIMEOUT, SECONDS);
+  }
+
+  @Test
+  @PerfTest(duration = 15000, threads = 1, warmUp = 5000)
+  public void fluxPull() throws Exception {
```
Comment (marianogonzalez) [inline tests/performance/src/test/java/org/mule/test/core/flow/FlowPerformanceTestCase.java]: add jdoc. It's not at all clear what is being tested here. This also seems as something we should be adding Allure annotations to.

File: tests/performance/src/test/java/org/mule/test/core/flow/RingBufferProcessingStrategyFactory.java [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+      super(schedulerStopper, muleContext);
+      this.cpuLightSchedulerSupplier = cpuLightSchedulerSupplier;
+    }
+
+    @Override
+    public Function<Publisher<Event>, Publisher<Event>> onPipeline(FlowConstruct flowConstruct,
+                                                                   Function<Publisher<Event>, Publisher<Event>> pipelineFunction,
+                                                                   MessagingExceptionHandler messagingExceptionHandler) {
+      return publisher -> from(publisher)
+          // TODO Work out why this fails with our scheduler
```
Comment (marianogonzalez) [inline tests/performance/src/test/java/org/mule/test/core/flow/RingBufferProcessingStrategyFactory.java]: attach a jira issue

File: tests/performance/src/test/java/org/mule/test/core/flow/FlowPerformanceTestCase.java (hunk 2 above)
Comment (dfeist) [inline tests/performance/src/test/java/org/mule/test/core/flow/FlowPerformanceTestCase.java]: `countDown` is not a Event consumer.

**Pull request merged**